### PR TITLE
Drop some FKs, trigger and ununsed indexes

### LINKED ISSUE #512: Record Parser unable to keep up
**Detailed Description**
In the performance test environment, 1.5K TPS was ran at 11 CST then pushed to 5K TPS at 3 CST. During this time the mirror node became increasingly unable to keep up, eventually becoming 7hrs behind in 23hr test.

**Actual Behavior**
Generate a load on mainnet and watch parser not keep up

**Expected Behavior**
Record parser can keep up with increased transactions per second that mainnet can handle.

**Environment:**
- Java: OpenJDK 11.0.2
- OS: Ubuntu 18.04
- Version: v0.5.2

**Additional Context**
PostgreSQL statistics show the inserts to `t_cryptotransferlists` and `t_transactions` are the heaviest used part of the system at the time.

```
mirror_node=> SELECT substring(query, 1, 50) AS short_query,
mirror_node->               round(total_time::numeric, 2) AS total_time,
mirror_node->               calls,
mirror_node->               round(mean_time::numeric, 2) AS mean,
mirror_node->               round((100 * total_time /
mirror_node(>               sum(total_time::numeric) OVER ())::numeric, 2) AS percentage_cpu
mirror_node-> FROM    pg_stat_statements
mirror_node-> ORDER BY total_time DESC
mirror_node-> LIMIT 20;
                    short_query                     | total_time  |   calls    | mean  | percentage_cpu
----------------------------------------------------+-------------+------------+-------+----------------
 INSERT INTO t_cryptotransferlists (consensus_times | 25789531.56 | 1161347973 |  0.02 |          68.57
 INSERT INTO t_transactions (fk_node_acc_id, memo,  | 10149207.22 |  232698039 |  0.04 |          26.99
 <insufficient privilege>                           |   599819.72 |      57614 | 10.41 |           1.59
 <insufficient privilege>                           |   577730.14 |      54292 | 10.64 |           1.54
 <insufficient privilege>                           |   262085.52 |      57614 |  4.55 |           0.70
 <insufficient privilege>                           |    39465.54 |    1139951 |  0.03 |           0.10
 <insufficient privilege>                           |    30767.96 |    1727118 |  0.02 |           0.08
 INSERT INTO t_contract_result (consensus_timestamp |    30511.47 |    1462108 |  0.02 |           0.08
 <insufficient privilege>                           |    25529.41 |      57614 |  0.44 |           0.07
 select entities0_.id as id1_3_0_, entities0_.auto_ |    22934.61 |    1461926 |  0.02 |           0.06
 select * from f_file_create($1, $2 )  as result    |    13087.12 |      53856 |  0.24 |           0.03
 <insufficient privilege>                           |    10791.66 |     938608 |  0.01 |           0.03
 <insufficient privilege>                           |    10605.67 |      57614 |  0.18 |           0.03
 select * from f_file_complete( $1, $2, $3 )  as re |     9218.32 |      27629 |  0.33 |           0.02
 <insufficient privilege>                           |     8764.41 |      57614 |  0.15 |           0.02
 <insufficient privilege>                           |     6438.64 |      57614 |  0.11 |           0.02
 update t_application_status set status_value=$1 wh |     4240.41 |      93368 |  0.05 |           0.01
 select applicatio0_.status_code as status_c1_0_0_, |     3256.72 |      86764 |  0.04 |           0.01
 insert into account_balances (consensus_timestamp, |     2593.74 |     114249 |  0.02 |           0.01
 <insufficient privilege>                           |     2375.97 |     115227 |  0.02 |           0.01
```

### PR DESCRIPTION
**Pull request by apeksharma**
Signed-off-by: Apekshit Sharma <apekshit.sharma@hedera.com>

**Detailed description**:
Changes to optimize write performance
- Drop foreign keys in t_transactions
- Drop foreign keys referencing t_transactions(consensus_timestamp) in t_cryptotransferlists, t_file_data,
-   t_livehashes and t_contract_result.
- Drop trigger in topic_message
- Drop unused indexes: idx_t_transactions_rec_file, idx__t_cryptotransferlist_amount

Removing trigger results in failure of `NotifyingTopicListenerTest`. So the change cascades to deletion of notify listener related code.

**Which issue(s) this PR fixes**:
Partly addresses #512

**Special notes for your reviewer**:
Not renaming the columns `fk_payer_acc_id`, `fk_node_acc_id`, `fk_cud_entity_id` and `fk_rec_file_id` to keep this change simple and sweet. If needed, those can be done as part of upcoming changes to how we write/update entities from importer.

**Checklist**
- [ ] Documentation added
- [ ] Tests updated

### CODE REVIEW

**Comment by apeksharma** [comment]
Only remaining FK in our db is
```ALTER TABLE t_entities
        ADD CONSTRAINT fk_ent_type_id FOREIGN KEY (fk_entity_type_id) REFERENCES t_entity_types (id) ON UPDATE CASCADE ON DELETE CASCADE;```
Everyone okay to delete that one too?

File: hedera-mirror-importer/src/main/resources/db/migration/V1.19.0__drop_fks_trigger_unused_indexes.sql [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+alter table t_cryptotransferlists
+    drop constraint if exists fk__t_transactions;
+alter table t_file_data
+    drop constraint if exists fk__t_transactions;
+alter table t_livehashes
+    drop constraint if exists fk__t_transactions;
+alter table t_contract_result
+    drop constraint if exists fk__t_transactions;
+
+drop trigger topic_message_trigger on topic_message;
```
Comment (steven-sheehy) [inline hedera-mirror-importer/src/main/resources/db/migration/V1.19.0__drop_fks_trigger_unused_indexes.sql]: Delete the related function as well

**Comment by apeksharma** [comment]
> idx_t_transactions_payer_id

**Comment by mike-burrage-hedera** [comment]
The node index is only used in ad-hoc queries.
If the other index goes away the "one transaction" query should use the "transaction_id" index instead.

**Comment by mike-burrage-hedera** [comment]
Actually sorry 600k uses on the node index scan - if it's still going up, it might be used by something. I couldn't find it in use in the rest-api, but might have missed something.

**Comment by mike-burrage-hedera** [comment]
I'd run some transactions rest-api calls to make sure, but if that node index is _actually_ needed, it's a pretty horrible one as it's selectivity would be only 1/13 of 100M+ rows...

**Comment by apeksharma** [comment]
Since those two indexes requires more exploration, i'd love if we can merge in this PR which is ready and do that task separately.

**Comment by mike-burrage-hedera** [comment]
If we're trying to measure the impact of removing indexes/fks that have an insert performance penalty that grows logarithmically with existing table row counts, then lets remove 2 more indexes on the high row count t_transactions table and not revisit the issue later.
It requires a simple test against /api/v1/transactions and /api/v1/transactions/:id to ensure they can be removed.

**Comment by apeksharma** [comment]
Yeah, don't worry, we'll do it. But that work doesn't have to be blocker for this PR which is useful and ready to go in. Can do it in tomorrow or early next week. Created issue to make sure it's done. (https://app.zenhub.com/workspaces/hedera-mirror-node-5d6fc4fca9be420001e7d223/issues/hashgraph/hedera-mirror-node/548).

**Comment by mike-burrage-hedera** [comment]
15 minutes of work now to call those rest-api requests, take the log output and run explain analyze before and after making the index drop.

**Comment by apeksharma** [comment]
If it's something i have to do, please let me do it at my leisure.
I'll have to run rest api perf test, log queries, analyze, run db perf test against baseline, etc.

If you want it *now*, please go ahead and do it.

**Comment by steven-sheehy** [comment]
Since we want to have an RC on Friday, we need to ensure that any change that goes in doesn't cause a regression. Removing an index can cause a large regression in REST API and we don't have test coverage to catch that. Better to do the quick check that the affected query runs in decent time after the index removed. This is normal part of testing any such change before it being accepted.

**Pull request merged**